Add Term Entry: JavaScript Strings: `.slice()` method (#651)
Codecademy/docs: content/javascript/concepts/strings/terms/slice/slice.md

---
Title: '.slice()'
Description: 'Returns a selected portion of a string.'
Subjects:
  - 'Web Development'
  - 'Computer Science'
Tags:
  - 'Elements'
  - 'Methods'
  - 'Strings'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

The `.slice()` method selects and returns a selected portion of a string.

## Syntax

```pseudo
string.slice(start);

string.slice(start, stop);
```

> - The `start` index is optional and begins at zero (inclusive). Default is 0.

Reviewer: ```suggestion
- The `start` index is optional and begins at zero (inclusive) by default.
```

> - The `stop` index is also optional and defaults to the length of the string (not inclusive). Default is last element.

Reviewer: ```suggestion
- The `stop` index is also optional and defaults to the length of the string (not inclusive).
```

## Example

> 

Reviewer: Like in the "Codebyte Example", let's include 1 or 2 sentences here about what is going on in the example snippet below.

```js

> let smoothie = 'mango strawberry pineapple';

Reviewer: ```suggestion
let smoothie = "mango strawberry pineapple";
```

Let's use double-quotes for JS strings (I'm pretty sure other similar use them as well).

let berry = smoothie.slice(6, 16);

> console.log(fruit);

Reviewer: Getting an error because we got rid of `fruit`. Let's make the following edit:
```suggestion
console.log(smoothie);
```

// Output: mango strawberry pineapple

console.log(berry);
// Output: strawberry
```

> 

Reviewer: For example snippets, let's include a short sentence that preview a \```shell snippet with the output:

```suggestion

The output will look like the following:

```

And then the snippet would look like this:

```shell
mango strawberry pineapple
strawberry
```

## Codebyte Example

> In the example below, the `animals` array is created. Then the `birds` array is created to extract the birds "eagles" and "owls". Lastly, the `birds` array is used to output a sentence.

Reviewer: ```suggestion
In the example below, an `animals` string is created. Next, the `.slice()` method is used to pick out substrings of certain animals, specifically birds:
```

---
Title: '.slice()'
Description: 'Returns a selected portion of a string.'
Subjects:
  - 'Web Development'
  - 'Computer Science'
Tags:
  - 'Elements'
  - 'Methods'
  - 'Arrays'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

> The JavaScript method `.slice()` returns a selected portion of a string.

Reviewer: ```suggestion
The `.slice()` method selects and returns a portion of a string.
```

## Syntax

```pseudo

> slice(start)
> slice(start, stop)

Reviewer: ```suggestion
string.slice(start);

string.slice(start, stop);
```

```

> - `start` utilizes zero based indexing and represents the starting index of the extraction (inclusive). This is optional and the default is 0.
> - `stop` utilizes zero based indexing and represents the ending index of the extraction (not inclusive). This is optional and the default is the rest of the array (`arr.length`).

Reviewer: ```suggestion
- The `start` index is optional and begins at zero (inclusive).
- The `stop` index is also optional and defaults to the length of the string (not inclusive).
```

## Example

```js
let fruits = ['Mango', 'Apple', 'Strawberry', 'Blueberry', 'Pineapple'];

let berries = fruits.slice(2, 4);

console.log(fruits);
// Output: ['Mango', 'Apple', 'Strawberry', 'Blueberry', 'Pineapple']

console.log(berries);
// Output: ['Strawberry', 'Blueberry']
```

## Codebyte Example

In the example below the animals array is created. Then the birds array is created to extract the birds "eagles" and "owls". Lastly, the birds array is used to output a sentence.

> ```codebyte/python

Reviewer: ```suggestion
```codebyte/javascript
```

---
Title: '.slice()'
Description: 'Returns a selected portion of a string.'
Subjects:
  - 'Web Development'
  - 'Computer Science'
Tags:
  - 'Elements'
  - 'Methods'

>   - 'Arrays'

Reviewer 2: ```suggestion
  - 'Strings'
```

Reviewer 2: This is directly correlated to @Dusch4593 's comment ☺️

CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

> The `.slice()` method selects and returns a portion of a string.

Reviewer 2: ```suggestion
The `.slice()` method selects and returns a selected portion of a string.
```

> The `.slice()` method selects and returns a portion of a string.

Reviewer 2: ```suggestion
The `.slice()` method selects and returns a selected portion of a string.
```

## Syntax

```pseudo
string.slice(start);

string.slice(start, stop);
```

> - The `start` index is optional and begins at zero (inclusive).

Reviewer 2: Can we include what the default value will be if nothing is specified?

> - The `start` index is optional and begins at zero (inclusive).

Reviewer 2: Can we include what the default value will be if nothing is specified?

- The `stop` index is also optional and defaults to the length of the string (not inclusive).

## Example

```js

> let fruits = ['Mango', 'Apple', 'Strawberry', 'Blueberry', 'Pineapple'];
> 
> let berries = fruits.slice(2, 4);
> 
> console.log(fruits);
> // Output: ['Mango', 'Apple', 'Strawberry', 'Blueberry', 'Pineapple']
> 
> console.log(berries);
> // Output: ['Strawberry', 'Blueberry']

Reviewer 2: ```suggestion
let smoothie = 'mango strawberry pineapple';

let berry = smoothie.slice(6, 16);

console.log(fruit);
// Output: mango strawberry pineapple

console.log(berry);
// Output: strawberry
```

```

## Codebyte Example

> In the example below the animals array is created. Then the birds array is created to extract the birds "eagles" and "owls". Lastly, the birds array is used to output a sentence.

Reviewer 2: ```suggestion
In the example below, the `animals` array is created. Then the `birds` array is created to extract the birds "eagles" and "owls". Lastly, the `birds` array is used to output a sentence.
```
```suggestion
In the example below the animals array is created. Then the birds array is created to extract the birds "eagles" and "owls". Lastly, the birds array is used to output a sentence.
```

> In the example below the animals array is created. Then the birds array is created to extract the birds "eagles" and "owls". Lastly, the birds array is used to output a sentence.

Reviewer 2: ```suggestion
In the example below, the `animals` array is created. Then the `birds` array is created to extract the birds "eagles" and "owls". Lastly, the `birds` array is used to output a sentence.
```
```suggestion
In the example below the animals array is created. Then the birds array is created to extract the birds "eagles" and "owls". Lastly, the birds array is used to output a sentence.
```

```codebyte/javascript

> let animals = ['elephants', 'eagles', 'owls', 'tigers'];
> 
> let birds = animals.slice(1, 3);
> 
> console.log(`${birds[0]} and ${birds[1]} are cool birds`);

Reviewer 2: ```suggestion
let animals = 'elephants, eagles, owls, and tigers';

console.log(`${animals.slice(11,17)} and ${animals.slice(19,23)} are cool birds`);
```

Reviewer 2: @denisCabrera I made some code example suggestions, although feel free to adjust as desired!

---
Title: '.slice()'
Description: 'Returns a selected portion of a string.'
Subjects:
  - 'Web Development'
  - 'Computer Science'
Tags:
  - 'Elements'
  - 'Methods'
  - 'Strings'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

The `.slice()` method selects and returns a selected portion of a string.

## Syntax

```pseudo
string.slice(start);

string.slice(start, stop);
```

- The `start` index is optional and begins at zero (inclusive) by default.
- The `stop` index is also optional and defaults to the length of the string (not inclusive).

## Example

In the example below, a `smoothie` string is created. Next, the `.slice()` method is used to extract substrings for the berry variable:

```js
let smoothie = "mango strawberry pineapple";

let berry = smoothie.slice(6, 16);

console.log(smoothie);

> // Output: mango strawberry pineapple

Reviewer 3: These comments are redundant with the output listed below. I'd remove them.

Author: I agree. Thank you
